### PR TITLE
Consider people in distance calculation

### PR DESCRIPTION
**Pull request by jedahan**
This should make similar items be more similar

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,105 +1,110 @@
...
-  "generic_desc_*"
-];
+  'generic_desc_*',
+  'curvy',
+  'vertical',
+  'diagonal',
+  'horizontal',
+  'light',
+  'line'
+]
```
Comment (jedahan) [inline server/app.js]: Added curvy, vertical, diagonal, horizontal, light and line

File: server/app.js [hunk 2]
```diff
@@ -109,98 +114,99 @@ const esClient = new elasticsearch.Client({
...
-      res.json(esRes);
+      res.json(esRes)
     }
-  });
-});
+  })
+})
+
+function getObjectDescriptors (objectID) {
+  const descriptorFields = MORE_LIKE_THIS_FIELDS.concat('people')
```
Comment (jedahan) [inline server/app.js]: hardcoded people, this distance calculation should maybe be in its own module in the future, and refactored...

File: server/app.js [hunk 3]
```diff
@@ -235,49 +241,57 @@ app.get('/api/related', (req, res) => {
...
 
-      const sortedDistances = distances.slice().sort();
+      const sortedDistances = distances.slice().sort()
 
-      const median = sortedDistances[Math.floor(distances.length/2)]
+      const quartiles = [
+        sortedDistances[Math.floor(1 * distances.length / 4)],
+        sortedDistances[Math.floor(2 * distances.length / 4)],
+        sortedDistances[Math.floor(3 * distances.length / 4)]
+      ]
```
Comment (jedahan) [inline server/app.js]: switched to quartiles, so we consider the bottom 25% and top 25% to increase diversity

File: server/app.js [hunk 4]
```diff
@@ -235,49 +241,57 @@ app.get('/api/related', (req, res) => {
...
+          const absoluteDistance = parseFloat(objectDescriptors[key]) - parseFloat(source[key])
+          return sum + (absoluteDistance * absoluteDistance)
         }, 0)
 
-        const normalized_distance =  distance / descriptor_keys.length
+        let normalizedDistance = distance / descriptorKeys.length
+
+        if (source.people !== objectDescriptors.people) {
+          normalizedDistance += 250
+        }
```
Comment (jedahan) [inline server/app.js]: we now add 250 for different artists

**Comment by erchaves** [comment]
👍

**Pull request merged**